tests/Fixer/LanguageConstruct/SingleSpaceAfterConstructFixerTest.php

```diff
@@ -0,0 +1,421 @@
+
+/**
+ * @author Andreas Möller <am@localheinz.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\LanguageConstruct\SingleSpaceAfterConstructFixer
+ */
+final class SingleSpaceAfterConstructFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @dataProvider provideInvalidConstructCases
+     *
+     * @param mixed $construct
+     */
+    public function testConfigureRejectsInvalidControlStatement($construct)
+    {
+        $this->expectException(InvalidFixerConfigurationException::class);
+
+        $this->fixer->configure([
+            'constructs' => [$construct],
+        ]);
+    }
+
+    /**
+     * @return array
+     */
+    public function provideInvalidConstructCases()
+    {
+        return [
+            'null' => [null],
+            'false' => [false],
+            'true' => [true],
+            'int' => [0],
+            'float' => [3.14],
+            'array' => [[]],
+            'object' => [new \stdClass()],
+            'unknown' => ['foo'],
+        ];
+    }
+
+    /**
+     * @dataProvider provideFixDefaultCases
+     *
+     * @param string      $expected
+     * @param null|string $input
+     */
+    public function testFix($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixDefaultCases()
+    {
+        return array_merge(
+            $this->provideFixWithBreakCases(),
+            $this->provideFixWithContinueCases(),
+            $this->provideFixWithContinueCases(),
+            $this->provideFixWithReturnCases(),
+            $this->provideFixWithThrowCases(),
+            $this->provideFixWithYieldCases()
+        );
+    }
+
+    /**
+     * @dataProvider provideFixWithBreakCases
+     *
+     * @param string      $expected
+     * @param null|string $input
+     */
+    public function testFixWithBreak($expected, $input = null)
+    {
+        $this->fixer->configure([
+            'constructs' => [
+                'break',
+            ],
+        ]);
+
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixWithBreakCases()
+    {
+        return [
+            [
+                '<?php break 9000; ?>',
+            ],
+            [
+                '<?php break 9000; ?>',
+                '<?php break  9000; ?>',
+            ],
+            [
+                '<?php break /* foo */9000; ?>',
+                '<?php break/* foo */9000; ?>',
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider provideFixWithContinueCases
+     *
+     * @param string      $expected
+     * @param null|string $input
+     */
+    public function testFixWithContinue($expected, $input = null)
+    {
+        $this->fixer->configure([
+            'constructs' => [
+                'continue',
+            ],
+        ]);
+
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixWithContinueCases()
+    {
+        return [
+            [
+                '<?php continue 9000; ?>',
```

Comment:
this is duplicated by next case

Reply:
How so?

I want to ensure that we have a single space, right?

Reply:
```php
            [
                '<?php return 9000;',
            ],
            [
                '<?php return 9000;',
                '<?php return  9000;',
            ],
```

first scenario will test
`<?php return 9000;` got not changed

2nd scenario will test
`<?php return 9000;` got not changed
`<?php return  9000;` got changed into `<?php return 9000;`

Reply:
Hmm, but doesn't the second scenario test that

```php
<?php return  9000;
```

is fixed to 

```php
<?php return 9000;
```

?

Reply:
if you have `expected + input` it test 2 things, as I said.
- `input` was fixed into `expected`
- `expected` is not modified when fixing it

Reply:
Ha, I see - wasn't aware of it.